Fakeroot test failures on Fedora 33 - /sys/fs/selinux remount

### Version of Singularity:

master


### Actual behavior

Fakeroot related build tests fail:

```
--- FAIL: TestE2E/PAR/BUILD (0.02s)
            --- FAIL: TestE2E/PAR/BUILD/issue_5690 (0.33s)
            --- FAIL: TestE2E/PAR/BUILD/issue_4837 (0.10s)
            --- FAIL: TestE2E/PAR/BUILD/issue_5315 (0.08s)
            --- FAIL: TestE2E/PAR/BUILD/from_local_image (11.52s)
                --- FAIL: TestE2E/PAR/BUILD/from_local_image/Fakeroot (0.42s)
                    --- FAIL: TestE2E/PAR/BUILD/from_local_image/Fakeroot/SIFToSIF (0.10s)
                    --- FAIL: TestE2E/PAR/BUILD/from_local_image/Fakeroot/SandboxToSIF (0.08s)
                    --- FAIL: TestE2E/PAR/BUILD/from_local_image/Fakeroot/LocalImage (0.08s)
                    --- FAIL: TestE2E/PAR/BUILD/from_local_image/Fakeroot/LocalImageLabel (0.07s)
                    --- FAIL: TestE2E/PAR/BUILD/from_local_image/Fakeroot/LocalImageSandbox (0.09s)
            --- FAIL: TestE2E/PAR/BUILD/definition (124.92s)
                --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot (1.32s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Files (0.08s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Env (0.07s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Pre (0.08s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Post (0.07s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/AppEnv (0.07s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/AppTest (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Setup (0.08s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/AppLabels (0.07s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/AppRun (0.07s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Empty (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Test (0.07s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Startscript (0.07s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Runscript (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Labels (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/AppHelp (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/AppFiles (0.07s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/Help (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/definition/Fakeroot/AppInstall (0.06s)
            --- FAIL: TestE2E/PAR/BUILD/from (182.33s)
                --- FAIL: TestE2E/PAR/BUILD/from/Root (181.76s)
                    --- FAIL: TestE2E/PAR/BUILD/from/Root/Yum (52.16s)
                --- FAIL: TestE2E/PAR/BUILD/from/Fakeroot (0.57s)
                    --- FAIL: TestE2E/PAR/BUILD/from/Fakeroot/BusyBox (0.14s)
                    --- FAIL: TestE2E/PAR/BUILD/from/Fakeroot/Debootstrap (0.12s)
                    --- FAIL: TestE2E/PAR/BUILD/from/Fakeroot/DockerURI (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/from/Fakeroot/DockerDefFile (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/from/Fakeroot/LibraryDefFile (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/from/Fakeroot/OrasURI (0.06s)
                    --- FAIL: TestE2E/PAR/BUILD/from/Fakeroot/Yum (0.06s)
```

Cause is a remount of /sys/fs/selinux :

```
=== RUN   TestE2E/PAR/BUILD/definition/Fakeroot/Env
    imgbuild.go:810: Running command "/usr/local/bin/singularity build --fakeroot --sandbox /tmp/stest.454226242/build-definition.405404814/container /tmp/TestTemplate-588783765"
    imgbuild.go:810: 
        "/usr/local/bin/singularity build --fakeroot --sandbox /tmp/stest.454226242/build-definition.405404814/container /tmp/TestTemplate-588783765" output:
        FATAL:   while remount /sys/fs/selinux read-only: operation not permitted
        
    imgbuild.go:810: got 255 as exit code and was expecting 0: exit status 255
        waiting for command "/usr/local/bin/singularity build --fakeroot --sandbox /tmp/stest.454226242/build-definition.405404814/container /tmp/TestTemplate-588783765"
        github.com/sylabs/singularity/e2e/internal/e2e.TestEnv.RunSingularity.func1
                github.com/sylabs/singularity/e2e/internal/e2e/singularitycmd.go:658
        testing.tRunner
                testing/testing.go:1127
        runtime.goexit
                runtime/asm_amd64.s:1374
```


### What OS/distro are you running

```
Fedora 33

$ uname -a
Linux dev-fedora.home.randomroad.net 5.9.8-200.fc33.x86_64 #1 SMP Tue Nov 10 21:58:19 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

--- a/internal/pkg/runtime/engine/fakeroot/engine_linux.go
+++ b/internal/pkg/runtime/engine/fakeroot/engine_linux.go
@@ -219,11 +219,15 @@ func (e *EngineOperations) StartProcess(masterConn net.Conn) error {
 		return fmt.Errorf("while parsing %s: %s", mountInfo, err)
 	}
 	for _, m := range mounts["/sys"] {
+		// In Linux <5.9 this is required so that in the chroot, selinux is seen as ro, i.e.
+		// disabled, and errors getting security labels do not occur.
+		// In 5.9 the remount will now fail, but it is not needed due to changes in label handling.
 		if m == selinuxMount {
 			flags := uintptr(syscall.MS_BIND | syscall.MS_REMOUNT | syscall.MS_RDONLY)
 			err = syscall.Mount("", selinuxMount, "", flags, "")
 			if err != nil {
-				return fmt.Errorf("while remount %s read-only: %s", selinuxMount, err)
+				sylog.Debugf("while remount %s read-only: %s", selinuxMount, err)
+				sylog.Debugf("note %s remount failure is expected on kernel 5.9+", selinuxMount)
 			}
 			break
 		}

